graph/config/cytoscape/cytoscape.go

```diff
@@ -438,3 +450,12 @@ func generateGroupCompoundNodes(appBox map[string][]*NodeData, nodes *[]*NodeWra
 		}
 	}
 }
+
+func calPrecisionDelta(rateVal float64) int {
+	delta := 0
+	if rateVal < 1 {
+		base := int(1 / rateVal)
+		delta = len(strconv.Itoa(base))
```

Comment:
is there way to achieve a manipulation on a number without converting it to string?

Reply:
That can only be done in a loop

Reply:
You could use `log10(...) + 1` to get the same value with only maths.